crates/chain/tests/test_tx_graph_conflicts.rs

```diff
@@ -591,10 +591,84 @@ fn test_tx_conflict_handling() {
+                },
+                TxTemplate {
+                    tx_name: "second",
+                    inputs: &[TxInTemplate::PrevTx("first", 0)],
+                    outputs: &[TxOutTemplate::new(900, Some(0))],
+                    ..Default::default()
+                },
+                TxTemplate {
+                    tx_name: "anchored",
+                    inputs: &[TxInTemplate::PrevTx("second", 0)],
+                    outputs: &[TxOutTemplate::new(800, Some(0))],
+                    anchors: &[block_id!(3, "D")],
+                    ..Default::default()
+                },
+            ],
+            exp_chain_txs: HashSet::from(["first", "second", "anchored"]),
+            exp_chain_txouts: HashSet::from([("first", 0), ("second", 0), ("anchored", 0)]),
+            exp_unspents: HashSet::from([("anchored", 0)]),
+            exp_balance: Balance {
+                immature: Amount::ZERO,
+                trusted_pending: Amount::ZERO,
+                untrusted_pending: Amount::ZERO,
+                confirmed: Amount::from_sat(800),
+            }
+        },
+        Scenario {
+            name: "transitively anchored txs should have priority over last seen",
+            tx_templates: &[
+                TxTemplate {
+                    tx_name: "root",
+                    inputs: &[TxInTemplate::Bogus],
+                    outputs: &[TxOutTemplate::new(10_000, Some(0))],
+                    anchors: &[block_id!(1, "B")],
+                    ..Default::default()
+                },
+                TxTemplate {
+                    tx_name: "last_seen_conflict",
+                    inputs: &[TxInTemplate::PrevTx("root", 0)],
+                    outputs: &[TxOutTemplate::new(9900, Some(1))],
+                    last_seen: Some(1000),
+                    ..Default::default()
+                },
+                TxTemplate {
+                    tx_name: "transitively_anchored_conflict",
+                    inputs: &[TxInTemplate::PrevTx("root", 0)],
+                    outputs: &[TxOutTemplate::new(9000, Some(1))],
+                    last_seen: Some(100),
+                    ..Default::default()
+                },
+                TxTemplate {
+                    tx_name: "anchored",
+                    inputs: &[TxInTemplate::PrevTx("transitively_anchored_conflict", 0)],
+                    outputs: &[TxOutTemplate::new(8000, Some(2))],
+                    anchors: &[block_id!(4, "E")],
+                    ..Default::default()
+                },
+            ],
+            exp_chain_txs: HashSet::from(["root", "transitively_anchored_conflict", "anchored"]),
+            exp_chain_txouts: HashSet::from([("root", 0), ("transitively_anchored_conflict", 0), ("anchored", 0)]),
+            exp_unspents: HashSet::from([("anchored", 0)]),
+            exp_balance: Balance {
+                confirmed: Amount::from_sat(8000),
+                ..Default::default()
+            }
+        }
```

Comment:
There seems to be a test missing for an anchored tx but not confirmed tx being canonical (or not). The orphaned block tests we have all are about conflicts.

P.S. this file is test_tx_graph_conflicts.rs but maybe it should just be `test_tx_graph_canonicalization.rs` .

Reply:
In terms of naming, I think conflicts and canonicalization are mostly synonymous. We don't need to canonicalize if we don't have conflicts.

Reply:
Test added.